File: 50 Coding Challenge Part I/TypeScript/Challenge17.ts
// Challenge 17: Calculate the sum of digits
// of a positive integer number

function sumDigits(num: number): number {
    let str: string = num.toString();
    let sum: number = 0;

    for (let char of str){
        let digits = parseInt(char);
        sum += digits;
    }

    return sum;
}

// Driver Method
console.log(sumDigits(12345));

File: 50 Coding Challenge Part I/dart/lib/challenge17.dart
/*  
  Coding Challenge 17: Calculate the sum of digits of a positive integer number

  @author: mofirojean
*/

int sumDigits(int num) {
  int sum = 0;

  if (num.isNegative) return 0;

  List<String> str = num.toString().split('');
  for (var s in str) {
    sum += int.parse(s);
  }

  return sum;
}
